pkg/domain/infra/tunnel/containers.go

```diff
@@ -92,7 +92,35 @@ func (ic *ContainerEngine) ContainerUnpause(ctx context.Context, namesOrIds []st
 
 func (ic *ContainerEngine) ContainerStop(ctx context.Context, namesOrIds []string, opts entities.StopOptions) ([]*entities.StopReport, error) {
 	reports := []*entities.StopReport{}
-	ctrs, rawInputs, err := getContainersAndInputByContext(ic.ClientCtx, opts.All, opts.Ignore, namesOrIds)
+
+	containersNamesOrIds := namesOrIds
+	if len(opts.Filters) > 0 {
+		containersNamesOrIds = []string{}
+		options := new(containers.ListOptions).WithFilters(opts.Filters).WithAll(true)
```

Comment:
I see that the same is done in podman-remote start as well but I really wonder if this is the right approach. Shouldn't we filter on the server side to prevent race conditions?
@vrothberg @mheon @rhatdan WDYT?

Reply:
Server-side seems safest